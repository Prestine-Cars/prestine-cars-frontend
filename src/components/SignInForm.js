// @ts-nocheck
import React, { useState } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { NavLink, useNavigate } from 'react-router-dom';
import { TailSpin } from 'react-loader-spinner';
import { login } from '../redux/actions/user';
import classes from './modules/SignInForm.module.css';
import logoImg from '../assets/logo.png';

const SignInForm = () => {
  const { Loader = false, loginError = null } = useSelector(
    (state) => state.user,
  );
  const [userLogin, setLogin] = useState({
    email: '',
    password: '',
  });
  const dispatch = useDispatch();
  const navigate = useNavigate();

  const handleLogin = (e) => {
    e.preventDefault();
    dispatch(login(userLogin, navigate));
  };

  const handleOnChange = (event) => {
    setLogin((prevState) => ({
      ...prevState,
      [event.target.name]: event.target.value,
    }));
  };
  return (
    <>
      <div className={classes['logo-image']}>
        <img src={logoImg} className={classes.logo} alt="Logo" />
      </div>
      <div className="container pt-4 mt-4">
        <div className="row d-flex justify-content-center align-items-center h-100">
          <div className="col-12 col-md-8 col-lg-6 col-xl-5">
            <div className={`card shadow-5-strong ${classes.sign_up_card}`}>
              <div className="card-body p-5 text-center">
                <div className="mb-md-2 mt-md-2 pb-5">
                  <h2 className="fw-bold mb-2 text-uppercase">Login</h2>
                  <p className="text-black-50 mb-5">
                    Please enter your login and password!
                  </p>
                  <form onSubmit={handleLogin}>
                    <div className={`${classes.loader}`}>
                      {Loader && (
                        <div>
                          <TailSpin
                            height="80"
                            width="80"
                            radius="3"
                            color="#98be20"
                            ariaLabel="tail-spin"
                            wrapperStyle
                            wrapperClass
                          />
                        </div>
                      )}
                    </div>

                    {loginError && (
                      <p>You have entered an invalid email/password</p>
                    )}
                    <div className="form-outline form-white mb-4">
                      <label htmlFor="email" className="form-label">
                        <input
                          onChange={handleOnChange}
                          type="email"
                          name="email"
                          placeholder="Email"
                          required
                          className="form-control form-control-lg"
                        />
                      </label>
                    </div>
                    <div className="form-outline form-white mb-4">
                      <label htmlFor="password" className="form-label">
                        <input
                          onChange={handleOnChange}
                          type="password"
                          name="password"
                          placeholder="Password"
                          required
                          className="form-control form-control-lg"
                        />
                      </label>
                    </div>
                    <p className="small mb-3 pb-lg-2">
                      <a className={`${classes.green_link}`} href="#!">
                        Forgot password?
                      </a>
                    </p>
                    <button
                      className={`btn text-white btn-block mb-2 ${classes.green_btn}`}
                      type="submit"
                    >
                      Login
                    </button>
                  </form>
                </div>

                <div>
                  <p className="mb-0">
                    Dont have an account?
                    {' '}
                    <NavLink className={`fw-bold ${classes.green_link}`} to="/signup">Sign Up</NavLink>
                  </p>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </>
  );
};

export default SignInForm;
