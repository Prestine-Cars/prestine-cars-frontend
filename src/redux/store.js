import { configureStore } from '@reduxjs/toolkit';
import userReducer from './reducers/user';
import cityReducer from './reducers/city';
import citiesReducer from './reducers/cities';
import carReducer from './reducers/car';

const store = configureStore({
  reducer: {
    user: userReducer,
    cities: citiesReducer,
    city: cityReducer,
    car: carReducer,
  },
});

export default store;
