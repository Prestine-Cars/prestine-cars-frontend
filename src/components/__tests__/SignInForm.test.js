import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import store from '../../redux/store';
import SignInForm from '../SignInForm';

describe('Test presence of Navbar', () => {
  test('should render Navbar component', () => {
    const signInForm = renderer
      .create(
        <BrowserRouter>
          <Provider store={store}>
            <SignInForm />
          </Provider>
        </BrowserRouter>,
      )
      .toJSON();
    expect(signInForm).toMatchSnapshot();
  });
});
