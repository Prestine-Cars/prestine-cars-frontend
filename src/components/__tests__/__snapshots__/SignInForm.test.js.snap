// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test presence of Sign In Form should render Sign In Form component 1`] = `
Array [
  <div
    className="logo-image"
  >
    <img
      alt="Logo"
      className="logo"
      src="logo.png"
    />
  </div>,
  <div
    className="container pt-4 mt-4"
  >
    <div
      className="row d-flex justify-content-center align-items-center h-100"
    >
      <div
        className="col-12 col-md-8 col-lg-6 col-xl-5"
      >
        <div
          className="card shadow-5-strong sign_up_card"
        >
          <div
            className="card-body p-5 text-center"
          >
            <div
              className="mb-md-2 mt-md-2 pb-5"
            >
              <h2
                className="fw-bold mb-2 text-uppercase"
              >
                Login
              </h2>
              <p
                className="text-black-50 mb-5"
              >
                Please enter your login and password!
              </p>
              <form
                onSubmit={[Function]}
              >
                <div
                  className="loader"
                />
                <div
                  className="form-outline form-white mb-4"
                >
                  <label
                    className="form-label"
                    htmlFor="email"
                  >
                    <input
                      className="form-control form-control-lg"
                      name="email"
                      onChange={[Function]}
                      placeholder="Email"
                      required={true}
                      type="email"
                    />
                  </label>
                </div>
                <div
                  className="form-outline form-white mb-4"
                >
                  <label
                    className="form-label"
                    htmlFor="password"
                  >
                    <input
                      className="form-control form-control-lg"
                      name="password"
                      onChange={[Function]}
                      placeholder="Password"
                      required={true}
                      type="password"
                    />
                  </label>
                </div>
                <p
                  className="small mb-3 pb-lg-2"
                >
                  <a
                    className="green_link"
                    href="#!"
                  >
                    Forgot password?
                  </a>
                </p>
                <button
                  className="btn text-white btn-block mb-2 green_btn"
                  type="submit"
                >
                  Login
                </button>
              </form>
            </div>
            <div>
              <p
                className="mb-0"
              >
                Dont have an account?
                 
                <a
                  className="fw-bold green_link"
                  href="/signup"
                  onClick={[Function]}
                >
                  Sign Up
                </a>
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
]
`;
